<?xml version="1.0" encoding="utf-8" ?>

<launch>

    <arg name="score_threshold" default="0.2"/>
    <arg name="iou_threshold" default="0.45"/>
    <arg name="encoding" default="rgb8"/>
    <arg name="device" default="cuda"/>
    <arg name="rviz" default="false"/>

    <!--  v5m  -->
    <arg name="config_path" default="$(find yolov5_ros)/submodules/yolov5/models/yolov5m.yaml"/>
    <arg name="weights_path" default="$(find yolov5_ros)/weights/yolov5m.pt"/>

    <arg name="input_image_topic" default="/head_camera/rgb/image_rect_color/compressed"/>
    <arg name="output_image_topic" default="/yolov5_ros/output/image/compressed"/>
    <arg name="output_bbox_topic" default="/yolov5_ros/output/bboxes"/>
    <arg name="service_topic" default="/yolov5_ros/service/detection"/>
    <arg name="action_topic" default="/yolov5_ros/action/detection"/>

    <node pkg="yolov5_ros" type="yolov5_node.py" name="yolov5_node" output="screen">
        <param name="score_threshold" type="double" value="$(arg score_threshold)"/>
        <param name="iou_threshold" type="double" value="$(arg iou_threshold)"/>
        <param name="encoding" type="str" value="$(arg encoding)"/>
        <param name="device" type="str" value="$(arg device)"/>

        <param name="weights_path" type="str" value="$(arg weights_path)"/>
        <param name="config_path" type="str" value="$(arg config_path)"/>

        <remap from="/input/image/compressed" to="$(arg input_image_topic)"/>
        <remap from="/input/activation" to="$(arg input_image_topic)"/>
        <remap from="/output/image/compressed" to="$(arg output_image_topic)"/>
        <remap from="/output/bboxes" to="$(arg output_bbox_topic)"/>
        <remap from="/service/detection" to="$(arg service_topic)"/>
        <remap from="/action/detection" to="$(arg action_topic)"/>
    </node>

  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find yolov5_ros)/config/rviz/layout.rviz"/>
  </group>

</launch>
